src/libraries/System.Private.CoreLib/src/System/AppContext.Browser.cs

```diff
@@ -0,0 +1,16 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+namespace System
+{
+    public static partial class AppContext
+    {
+        private static string GetBaseDirectoryCore()
+        {
+            // GetEntryAssembly().Location returns an empty string for wasm
+            // Until that can be easily changed, work around the problem here.
```

Comment:
How is this working around `GetEntryAssembly().Location` returning empty string?

Would it make more sense for `BaseDirectory` to be empty string as well when there is not a file system?

Reply:
Perhaps the wording is incorrect.  AppContext.BaseDirectory is returning "", which is causing a bunch of tests to fail.  I feel returning "/" isn't any less wrong than "".  The benefit of "/" being that it works within other parts of libraries.

Reply:
I think that returning "/" is more wrong than "".

If we were optimizing for making maximum tests to pass, we would also change Assembly.Location to return `/<assemblyname>.dll` that would be even more wrong.

What are the tests that fail before this change and pass with this change?

Reply:
I’ll follow up with a list.

Since the emscripten vfs mounts at /, I think my change does make sense.
https://emscripten.org/docs/porting/files/file_systems_overview.html

Reply:
Do the application binaries show up as files at  `/` ?

The docs for [AppContext.BaseDirectory](https://docs.microsoft.com/en-us/dotnet/api/system.appcontext.basedirectory) say: Gets the pathname of the base directory that the assembly resolver uses to probe for assemblies.

Reply:
Do the types exercised by there tests actually work on Wasm Browser? E.g. can I add configuration file to my .csproj file and then get the file picked up by the default `IFileProvider` at runtime?

Reply:
Good question.  If you look at the test results, we're likely at least somewhat functional.  

https://gist.github.com/steveisok/370622cdb0c41ff2bc3b439c730484e4

Reply:
I think we should have issues opened on making sure that the end-to-end works for these libraries. More passing tests does not necessarily mean that these libraries are actually usable by customers.

The linker analysis for single file unfriendly APIs (https://github.com/dotnet/runtime/issues/38405) should help us get a better handle on this.

If you believe that returning `/` here is going to help to make the affected scenarios work end-to-end, I can live with it.